### PR TITLE
Feature /  Swap and Bridge controller

### PR DESCRIPTION
**Pull request by sonytooo**
Implements the  the Swap and Bridge controller that is responsible for managing the state and logic related to swapping and bridging tokens across different networks. Key responsibilities:
- Initially setting up the swap and bridge form with the necessary data.
- Managing form state for token swap and bridge operations (including user preferences).
- Fetching and updating token lists (from and to).
- Fetching and updating quotes for token swaps and bridges.
- Manages token active routes

### CODE REVIEW

File: src/controllers/eventEmitter/eventEmitter.ts [hunk 1]
```diff
@@ -100,9 +108,7 @@ export default class EventEmitter {
           // Silence this error in prod to avoid displaying wired error messages.
           // The only benefit of displaying it is for devs to see when an action is dispatched twice.
           // TODO: If this happens on PROD, ideally we should get an error report somehow somewhere.
-          process.env.APP_ENV === 'production' && process.env.IS_TESTING !== 'true'
-            ? 'silent'
-            : 'minor',
+          errorLevel,
```
Comment (superKalo) [inline src/controllers/eventEmitter/eventEmitter.ts:111]: You moved the logic up, but left the comment explaining the reasoning here 👀 , maybe move the comment too?

Comment (superKalo) [reply src/controllers/eventEmitter/eventEmitter.ts]: Fixed in https://github.com/AmbireTech/ambire-common/pull/1005/commits/76b5c28f75141ba1245ee80765f65ee73ec7e757

File: src/services/address/address.ts [hunk 2]
```diff
@@ -20,4 +22,12 @@ const isHumanizerKnownTokenOrSmartContract = (
...
 }
 
-export { isValidAddress, isHumanizerKnownTokenOrSmartContract }
+const formatNativeTokenAddressIfNeeded = (address: string) => {
+  if (getAddress(address) === '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') {
+    return '0x0000000000000000000000000000000000000000'
+  }
+
+  return address
+}
```
Comment (superKalo) [inline src/services/address/address.ts:31]: TODO (for me): this is probably no longer needed after normalizing the incoming tokens. Double-check before mergeing!

File: src/controllers/dapps/dapps.ts [hunk 3]
```diff
@@ -113,6 +113,11 @@ export class DappsController extends EventEmitter {
   addDapp(dapp: Dapp) {
     if (!this.isReady) return
 
+    if (dapp.url.includes(chrome.runtime.id)) {
+      // eslint-disable-next-line no-param-reassign
+      dapp.url = chrome.runtime.id
+    }
```
[the remark this answers is not in the document]
Comment (sonytooo) [reply src/controllers/dapps/dapps.ts]: removed it cause no longer needed

File: src/services/address/address.ts (hunk 2 above)
Comment (superKalo) [reply src/services/address/address.ts]: Yep, removed in https://github.com/AmbireTech/ambire-common/pull/1005/commits/ac5f55c63b299f75da65108f3cee7a47e78c3bdd

**Review by superKalo — APPROVED** [review]

**Pull request merged**